bson/bsoncodec/struct_codec.go

```diff
@@ -127,12 +127,16 @@ func (sc *StructCodec) EncodeValue(r EncodeContext, vw bsonrw.ValueWriter, val r
 
 		encoder := desc.encoder
 
-		iszero := sc.isZero
-		if iz, ok := encoder.(CodecZeroer); ok {
-			iszero = iz.IsTypeZero
+		var isZero bool
```

Comment:
Code changes LGTM. Can you add a comment describing why we're doing these checks rather than just calling `sc.isZero`?

Reply:
done